ghedesigner/main.py

```diff
@@ -0,0 +1,272 @@
+import logging
+from json import loads
+from pathlib import Path
+from sys import exit, stderr
+
+import click
+from jsonschema import ValidationError
+
+from ghedesigner import VERSION
+from ghedesigner.enums import BHPipeType, DesignGeomType
+from ghedesigner.ghe.manager import GroundHeatExchanger
+from ghedesigner.heat_pump import HeatPump
+from ghedesigner.utilities import write_idf
+from ghedesigner.validate import validate_input_file
+
+logging.basicConfig(level=logging.WARN, format="%(message)s", datefmt="[%X]")
+logger = logging.getLogger(__name__)
+
+
+def _run_manager_from_cli_worker(input_file_path: Path, output_directory: Path) -> int:
+    """
+    Worker function to run simulation.
+
+    :param input_file_path: path to input file. Input file must exist.
+    :param output_directory: path to write output files. Output directory must be a valid path.
+    """
+
+    # validate inputs against schema before doing anything
+    if validate_input_file(input_file_path) != 0:
+        return 1
+
+    inputs = loads(input_file_path.read_text())
+
+    # Read in all the inputs into small dicts
+    # it is possible to define multiple fluids, GHEs, and boreholes in the inputs, I'm just taking the first for now
+    input_file_version: int = inputs["version"]
```

Comment:
Check input file version ID.  Could be useful later.

Reply:
Agreed, this could be useful if we change the input structure later.